membership_directors_board/models/membership_directors_board.py

```diff
@@ -0,0 +1,106 @@
+
+from openerp import models, fields, api
+
+
+class MembershipDirectorsBoard(models.Model):
+    _name = 'membership.directors.board'
+
+    @api.one
+    def _compute_president(self):
+        position_president_id = self.env.ref(
+            'membership_directors_board.'
+            'membership_directors_board_position_president')
+        member_obj = self.env['membership.directors.board.member']
+        president = member_obj.search(
+            [('governing_id', '=', self.id),
+             ('position_id', '=', position_president_id.id),
+             ('date_end', '=', False)], limit=1)
+        self.president_id = president.partner_id.id
+
+    @api.one
+    def _compute_secretary(self):
+        position_secretary_id = self.env.ref(
+            'membership_directors_board.'
+            'membership_directors_board_position_secretary')
+        member_obj = self.env['membership.directors.board.member']
+        secretary = member_obj.search(
+            [('governing_id', '=', self.id),
+             ('position_id', '=', position_secretary_id.id),
+             ('date_end', '=', False)], limit=1)
+        self.secretary_id = secretary.partner_id.id
+
+    name = fields.Char('Directors Board', required=True)
+    date_start = fields.Date('Init Date')
+    date_end = fields.Date('Finish Date')
+    note = fields.Text('Description')
+    members_ids = fields.One2many(
+        comodel_name='membership.directors.board.member',
+        inverse_name='directors_board_id', string='Members', required=True)
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'res.partner'))
+    president_id = fields.Many2one(
+        compute='_compute_president', comodel_name='res.partner',
+        string='President')
+    secretary_id = fields.Many2one(
+        compute='_compute_secretary', comodel_name='res.partner',
+        string='Secretary')
+
+    @api.model
+    def delete_all_tags(self, category):
+        partner_obj = self.env['res.partner']
+        partners = partner_obj.search(
+            [('category_id', 'in', category.ids)])
+        partners.write({'category_id': [(3, category.id)]})
+
+    @api.one
+    def update_directors_board_tag(self):
+        if self.company_id.membership_directors_board_id.id == self.id:
+            category = self.env.ref(
+                'membership_directors_board.membership_directors_board_tag')
+            self.delete_all_tags(category)
+            partner_obj = self.env['res.partner']
+            partner_ids = []
+            for member in self.members_ids.filtered(lambda r: not r.date_end):
+                partner_ids.append(member.partner_id.id)
+            new_partners = partner_obj.search([('id', 'in', partner_ids)])
+            new_partners.write({'category_id': [(4, category.id)]})
+        return True
+
+    @api.one
```

Comment:
Don't serialize write when there's more than one record to write. Think your method in multi-record environment.